src/fileseq/filesequence.py

```diff
@@ -434,3 +434,14 @@ def getPaddingChars(num):
             return "#" * (num / 4)
         else:
             return "@" * num
+
+    @staticmethod
+    def getPaddingNum(chars):
+        """
+        Given a supported group of padding characters, return the amount of padding.
+        :param chars: a supported group of padding characters (str)
+        :return: int
+        """
+        mapping = {'@': 1, '#': 4}
+
+        return sum([mapping[char] for char in chars])
```

Comment:
If you want to make it raise, then that would be acceptable. But it should
be an appropriate type of exception that is handled, and documented, and
not an obscure bubble up from the char not existing in a map.

On Sun, 9 Aug 2015 9:15 PM matthewkapfhammer <notifications@github.com>
wrote:

> In src/fileseq/filesequence.py
> <https://github.com/sqlboy/fileseq/pull/32#discussion_r36585901>:
>
> > @@ -434,3 +434,14 @@ def getPaddingChars(num):
> >              return "#" * (num / 4)
> >          else:
> >              return "@" * num
> > +
> > +    @staticmethod
> > +    def getPaddingNum(chars):
> > +        """
> > +        Given a supported group of padding characters, return the amount of padding.
> > +        :param chars: a supported group of padding characters (str)
> > +        :return: int
> > +        """
> > +        mapping = {'@': 1, '#': 4}
> > +
> > +        return sum([mapping[char] for char in chars])
>
> Yep. That would be preferable. Personally, I would expect the method to
> raise if unsupported characters were passed in rather than just ignore
> them.
>
> Just noticed that I should also be using PAD_MAP from the constants
> module.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sqlboy/fileseq/pull/32/files#r36585901>.
>